cmd/client/host.go

```diff
@@ -109,13 +109,20 @@ func (h *Host) logEntry(remoteID peer.ID) *log.Entry {
 
 // Bootstrap connects this host to bootstrap peers.
 func (h *Host) Bootstrap(ctx context.Context) error {
+	errCount := 0
+	var lastErr error
 	for _, bp := range h.bpAddrInfos {
 		log.WithField("remoteID", util.FmtPeerID(bp.ID)).Info("Connecting to bootstrap peer...")
 		if err := h.Connect(ctx, bp); err != nil {
-			return errors.Wrap(err, "connecting to bootstrap peer")
+			log.Warn("Error connecting to bootstrap peer", bp)
+			lastErr = errors.Wrap(err, "connecting to bootstrap peer")
 		}
 	}
 
+	if errCount == len(h.bpAddrInfos) {
```

Comment:
The punchr client needs to know its own observed addresses and if I see that correctly we need at least 4 confirmations from remote peers?

https://github.com/libp2p/go-libp2p/blob/6631bfb01100b730b1230d9b9ace43a8af4f5e5d/p2p/protocol/identify/obsaddr.go#L25

So I'd suggest:

```suggestion
	if len(h.bpAddrInfos) - errCount < identify.ActivationThresh {
```

but I'm absolutely not sure about this.

Reply:
I don't think we need this since we are already blocking until we know our observed address: https://github.com/dennis-tra/punchr/blob/main/cmd/client/host.go#L126. So if this is below the activation threshold in identify that will timeout and fail anyways.

Also I think it's possible we connect to only a single bootstrap node, and discover more nodes via the bootstrap node that can tell us our observed addr.

Reply:
Sounds good! :)